### PR TITLE
Fixes for issue #2998

### LINKED ISSUE #2998: helm dependency update fails after helm upgrade
Hi

I'm using Helm from Concourse pipeline I just upgraded Helm to 2.6.1 (from 2.4.2) and my pipeline doesn't work anymore.
The culprit line seems to be https://github.com/kubernetes/helm/blob/v2.6.1/pkg/downloader/manager.go#L220
```console
$ helm dependency update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "ci" chart repository
...Successfully got an update from the "local" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈Happy Helming!⎈
Error: Unable to move current charts to tmp dir: rename /tmp/build/put/staging-helm/cvhub-api/charts /tmp/build/put/staging-helm/cvhub-api/tmpcharts: invalid cross-device link
```
I cannot change the path used by Concourse  but if I do the same command manually on my mac from my home directory (i.e not from /tmp) it works:
```console
% ~/Downloads/helm dependency update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "ci" chart repository
...Successfully got an update from the "local" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Downloading elasticsearch-singlenode from repo http://127.0.0.1:8879
±|develop ✘| cvhub-api
% pwd
/Users/nsitbon/github/cvhub-api/scripts/helm/cvhub-api
```
As far as I understand the error and the related code it seems that you should not use Rename rather you should use Move.

Any help would be appreciated.

Best regards.

### PR DESCRIPTION
**Pull request by ysh7**
This PR includes changes mentioned on #5038 required for merging it.

closes #2998

### CODE REVIEW

**Comment by bacongobbler** [comment]
would you mind rebasing against master? That CI failure was due to a coreutils update that broke some CI scripts. Updating to master will get you the fixed scripts. Thanks

**Comment by ysh7** [comment]
Rebase done, tests are passing :fireworks:

File: third_party/dep/fs/fs.go [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Copyright The Helm Authors.
```
Comment (bacongobbler) [inline third_party/dep/fs/fs.go]: This has to be removed as we don't own copyright over this software. Can you please remove these "Copyright The Helm Authors" statements and implement the same exclusion rules for `validate-license.sh` as done for Helm 3 in #6609?

Also, if you wouldn't mind creating the `internal` package and move this over to `internal/third_party` as done in #6609, that'd be great. Thank you.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by bacongobbler** [comment]
ping

File: third_party/dep/fs/fs.go (hunk 1 above)
Comment (ysh7) [reply third_party/dep/fs/fs.go]: Sure

**Comment by bacongobbler** [comment]
We just switched branches over. Helm 2 development is on the `dev-v2` branch, and Helm 3 development is on the `master` branch. Would you mind re-opening this PR against the correct branch once this is ready for review again? Thanks.

**Pull request closed without merging**